ERCs/eip-5732/contracts/FiftyFifty.sol

```diff
@@ -0,0 +1,176 @@
+    /// @dev The committer will only ever be incentivized to call this function if they win,
+    ///     because otherwise they can will let guesser call timeout and pay gas for it.
+    function reveal(
+        uint256 gameId,
+        bool choice,
+        bytes32 salt
+    ) external;
+
+    /// @notice Allows the guesser to call timeout if the committer doesn't reveal in time.
+    /// @param gameId The id of the game to timeout.
+    function timeoutBet(uint256 gameId) external;
+
+    /// @notice Returns the total number of games.
+    /// @return count The total number of games.
+    function getGameCount() external view returns (uint256 count);
+
+    /// @notice Returns the game with the given ID.
+    /// @param gameId The ID of the game to get.
+    /// @return game The game with the given ID.
+    function getGame(uint256 gameId) external view returns (Game memory game);
+}
+
+contract FiftyFifty is IFiftyFifty {
+    uint256 private gameCount = 0;
+    mapping(uint256 => Game) private games;
+
+    /// @inheritdoc IFiftyFifty
+    function commit(bytes32 _commitment) external payable {
+        require(msg.value > 0, "cannot bet with 0 value");
+
+        gameCount++;
+
+        games[gameCount].committer = payable(msg.sender);
+        games[gameCount].commitment = _commitment;
+        games[gameCount].bet = msg.value;
+
+        emit BetPlaced(gameCount, msg.value, msg.sender, _commitment);
+    }
+
+    /// @inheritdoc IFiftyFifty
+    function cancelBet(uint256 _gameId) external {
+        require(games[_gameId].ended == false, "game already ended");
+        require(msg.sender == games[_gameId].committer, "cancellable only by committer");
+        require(games[_gameId].guesser == address(0), "cancellable only if guesser not set");
+
+        games[_gameId].ended = true;
+        games[_gameId].committer.transfer(games[_gameId].bet);
+
+        emit GameEnded(_gameId, GameEndReason.CANCEL, games[_gameId].committer);
+    }
+
+    /// @inheritdoc IFiftyFifty
+    function matchBet(uint256 _gameId, bool _choice) public payable {
+        require(games[_gameId].ended == false, "game already ended");
+        require(games[_gameId].guesser == address(0), "may only place bet if nobody else has");
+        require(msg.value == games[_gameId].bet, "bet amount does not match");
+
+        games[_gameId].guesser = payable(msg.sender);
+        games[_gameId].guess = _choice;
+        games[_gameId].expiration = block.timestamp + 7 days;
+
+        emit BetMatched(_gameId, _choice, msg.sender);
+    }
+
+    /// @inheritdoc IFiftyFifty
+    function reveal(
+        uint256 _gameId,
+        bool _choice,
+        bytes32 _salt
+    ) public {
+        require(games[_gameId].ended == false, "game already ended");
+        require(
+            games[_gameId].guesser != address(0),
+            "cannot reveal before anyone has matched bet"
+        );
+        require(
```

Comment:
This ensures `Reveal` is only allowed to be called by the creator of a single game. 
What does this game do to prevent / incentivize a gamer create to later reveal the result?
(good enough as a ref-impl for erc5732, but I just start to feel fasinated by this game you design and think it may have the potential to become a real DeGame)

Reply:
> This ensures Reveal is only allowed to be called by the creator of a single game.

Yes, or at least with the assurance that only the creator would know the secret salt. We could also hash the `msg.sender` in the commitment to enforce the same sender, but not sure if that is worth it.

> What does this game do to prevent / incentivize a gamer create to later reveal the result?

There is none! That's why there is an expiration timeout. In a typical game, if the *committer* had their choice guessed (and lost), they will not be incentivization to ever reveal, and therefor the guesser will have to collect the winnings via expiration timeout.

> (good enough as a ref-impl for erc5732, but I just start to feel fasinated by this game you design and think it may have the potential to become a real DeGame)

Thanks, I do plan to do more polishing (and some bitpacking for gas efficiency) for this ERCref. I think it could potentially have a wider use-case than just a fun DeGame, rather this could be used in even a serious situation where an on-chain 50/50 guess needed.